src/components/UserSelector.tsx

```diff
@@ -1,16 +1,69 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cl from 'classnames';
+import { getUsers } from '../api/users';
+import { User } from '../types/User';
+import { Loader } from './Loader';
+
+type Props = {
+  currentUser: User | null;
+  setCurrentUser: (u: User | null) => void;
+  hasError: boolean;
+  setHasError: (h: boolean) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  currentUser,
+  setCurrentUser,
+  hasError,
+  setHasError,
+}) => {
+  const [users, setUsers] = useState<User[]>();
+  const [isFocusDropdown, setIsFocusDropdown] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getUsers()
+      .then(usersFromServer => {
+        setUsers(usersFromServer);
+      })
+      .catch(() => setHasError(true))
+      .finally(() => setIsLoading(false));
+  }, []); // eslint-disable-line
+
+  const selectUser = (user: User) => {
+    setCurrentUser(user);
+    setIsFocusDropdown(false);
+  };
+
+  const handleBlur = (e: React.FocusEvent<HTMLButtonElement>) => {
+    if (e.relatedTarget?.className === 'dropdown-item') {
+      return;
+    }
+
+    setIsFocusDropdown(false);
+  };
+
+  const handleClick = () => {
+    setIsFocusDropdown(!isFocusDropdown);
+  };
 
-export const UserSelector: React.FC = () => {
   return (
-    <div data-cy="UserSelector" className="dropdown is-active">
+    <div
+      data-cy="UserSelector"
+      className={cl('dropdown', { 'is-active': isFocusDropdown })}
+    >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={handleClick}
+          onBlur={handleBlur}
         >
-          <span>Choose a user</span>
+          <span>{currentUser ? currentUser.name : 'Choose a user'}</span>
```

Comment:
Try to make destructuring assignment with currentUser object

Reply:
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/103579437/7f70e1bf-1fd3-42eb-b2ec-5e314a541c75)
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/103579437/56455ea2-8d20-43f9-975e-1d0d9915dae1)